addon/-private/serializers/json-api-serializer.js

```diff
@@ -109,11 +109,25 @@ const JSONAPISerializer = JSONSerializer.extend({
     if (Ember.typeOf(documentHash.data) === 'object') {
       documentHash.data = this._normalizeResourceHelper(documentHash.data);
     } else if (Ember.typeOf(documentHash.data) === 'array') {
-      documentHash.data = documentHash.data.map(this._normalizeResourceHelper, this);
+      let ret = new Array(documentHash.data.length);
+
+      for (let i = 0, l = documentHash.data.length; i < l; i++) {
+        let data = documentHash.data[i];
+        ret[i] = this._normalizeResourceHelper(data);
+      }
+
+      documentHash.data = ret;
     }
 
     if (Ember.typeOf(documentHash.included) === 'array') {
```

Comment:
i wonder if `Array.isArray(documentHash.data)` is sufficient (As it is much faster)